app-shared/src/commonMain/kotlin/io/github/droidkaigi/confsched/KaigiApp.kt

```diff
@@ -29,26 +32,28 @@ fun KaigiApp() {
             .build()
     }
     SwrClientProvider(SwrCachePlus(SwrCacheScope())) {
-        val subscription = rememberSubscription(
-            key = appGraph.createSettingsScreenContext().settingsSubscriptionKey,
-            select = { it.useKaigiFontFamily },
-        )
-        val kaigiFontFamily = if (subscription.isSuccess) {
-            when (subscription.data) {
-                KaigiFontFamily.ChangoRegular -> changoFontFamily()
-                KaigiFontFamily.RobotoRegular -> robotoRegularFontFamily()
-                KaigiFontFamily.RobotoMedium -> robotoMediumFontFamily()
-                KaigiFontFamily.SystemDefault -> null
-                null -> null
-            }
-        } else {
-            null
-        }
-
-        KaigiTheme(fontFamily = kaigiFontFamily) {
+        KaigiTheme(fontFamily = rememberKaigiFontFamily()) {
             Surface {
                 KaigiAppUi()
             }
         }
     }
 }
+
+@OptIn(ExperimentalSoilQueryApi::class)
+@Composable
+context(appGraph: AppGraph)
+private fun rememberKaigiFontFamily(): FontFamily? {
+    val subscription = rememberSubscription(
+        key = appGraph.createSettingsScreenContext().settingsSubscriptionKey,
+        select = { it.useKaigiFontFamily },
+    )
+    return subscription.reply.map {
+        when (it) {
+            KaigiFontFamily.ChangoRegular -> changoFontFamily()
+            KaigiFontFamily.RobotoRegular -> robotoRegularFontFamily()
+            KaigiFontFamily.RobotoMedium -> robotoMediumFontFamily()
+            KaigiFontFamily.SystemDefault -> null
+        }
+    }.getOrNull()
```

Comment:
Just a simple question out of curiosity 🙏

Since subscription.data is accessible regardless of the state (e.g., success), I was wondering if it’s fine to do something like this:
```kotlin
return when (subscription.data) {
    KaigiFontFamily.ChangoRegular -> changoFontFamily()
    KaigiFontFamily.RobotoRegular -> robotoRegularFontFamily()
    KaigiFontFamily.RobotoMedium -> robotoMediumFontFamily()
    KaigiFontFamily.SystemDefault -> null
    null -> null
}
```
Is there any difference if we use reply instead? I’m just curious. 👀

Reply:
Using data is perfectly fine 👌 

With Reply, you can write processing that targets the returned value like an Optional type. This means that when it's unresolved (such as during loading), it becomes Reply.None.

For this use case, your suggested approach seems better 🙏

Reply:
fixed: https://github.com/DroidKaigi/conference-app-2025/pull/619/commits/0a44a02e68a1881db8ee73b44ad99cb054dee1e5